devito/builtins.py

```diff
@@ -53,6 +54,131 @@ def smooth(f, g, axis=None):
         dv.Operator(dv.Eq(f, g.avg(dims=axis)), name='smoother')()
 
 
+def gaussian_smooth(f, sigma=1, _order=4, mode='reflect'):
+    """
+    Gaussian smooth function.
+    """
+    class DataDomain(dv.SubDomain):
+
+        name = 'datadomain'
+
+        def __init__(self, lw):
+            super(DataDomain, self).__init__()
+            self.lw = lw
+
+        def define(self, dimensions):
+            return {d: ('middle', self.lw, self.lw) for d in dimensions}
+
+    def fill_data(g, f, dim, lw, mode):
```

Comment:
isn't this basically `model/initialize_function`? Would a be lot shorter/more readable to reuse it IMO

Reply:
Will take a look, but I would have thought `model/initialize_function` can't do the `'reflect'` conditions?

Reply:
I didn't put it because didn't need it but it's just a small change in the rhs of the equation. Currently is `Eq(u(d), u(npml)` can just do `Eq(u(d), u(nbpm+d)` or something like that and you got the reflect

Reply:
Oki, will check it out.

Reply:
is it worth promoting initialize_function to a builtin ?